tests/unit/jobserver/test_middleware.py

```diff
@@ -85,6 +90,16 @@ def test_client_ip_middleware_proxied(rf):
     middleware(request)
     assert request.backend is None
 
+    # test spoofing header doesn't work
+    # attacker attempts to spoof both the TPP address & the proxy address
+    request = rf.get(
```

Comment:
Good call, this was designed to defend against this, but didn't include the test.

For just a single proxy, the last address wouldn't be there normally

Reply:
removed the last address